runtime/compiler/env/VMJ9.cpp

```diff
@@ -2488,8 +2491,10 @@ TR_J9VMBase::getClassSignature(TR_OpaqueClassBlock * clazz, TR_Memory * trMemory
    for (i = 0; i < numDims; i++)
       sig[i] = '[';
    if (* name != '[')
-      sig[i++] = 'L';
-   memcpy(sig+i, name, len);
+      if (isValueTypeClass(myClass) || hasUnflattenedFlattenableFields(myClass))
```

Comment:
As above, `Q` should be used only for a value type class - not for ordinary identity classes that have value type fields.  This should only check `isValueTypeClass`.